packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -257,7 +257,7 @@ class NumberInput extends Component {
           if (useControlledStateWithValue) {
             onChange(evt, { value });
           } else if (onChange) {
-            onChange(evt);
+            onChange(evt, { value: Number(value) });
```

Comment:
should this return the same params as `handleArrowClick`?

Reply:
@jnm2377 to answer [your question](https://github.com/carbon-design-system/carbon/pull/8243#discussion_r612565905) they should both return the same params